.github/workflows/test.yml

```diff
@@ -58,6 +58,12 @@ jobs:
           command: test
           args: --locked
 
+      - name: All Features
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --all-features
```

Comment:
This workflow step appears to run all the same tests as the previous workflow step, _plus_ the normally ignored features. Is there a way to specify that _only_ the normally ignored features are run? That way, there won't be duplicated test runs (whether that even adds any real overhead is unknown).

Reply:
I'm not sure what you mean by this? How would you run only the normally ignored features? Changing features might introduce arbitrary code changes that could fail any test.

Reply:
The `Test` step in the `test-matrix` job shows the following output:

```sh
C:\Users\runneradmin\.cargo\bin\cargo.exe test --locked
   Compiling phylum-cli v3.5.0 (D:\a\cli\cli\cli)
   Compiling xtask v0.1.0 (D:\a\cli\cli\xtask)
    Finished test [unoptimized + debuginfo] target(s) in 21.99s
     Running unittests src\lib.rs (target\debug\deps\phylum_cli-e181392bf4c854d8.exe)

running 46 tests
test api::tests::create_client ... ok
test api::tests::get_job_status ... ok
test api::tests::get_package_details ... ok
---TRIMMED---
test types::tests::test_risk_level_ordering ... ok
test commands::parse::tests::it_can_identify_lock_file_types ... ok

test result: ok. 46 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.93s

     Running unittests src\bin\phylum.rs (target\debug\deps\phylum-644a[12](https://github.com/phylum-dev/cli/runs/6580368267?check_suite_focus=true#step:8:13)93ac5782ec.exe)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests\cmd.rs (target\debug\deps\cmd-b561af1e[20](https://github.com/phylum-dev/cli/runs/6580368267?check_suite_focus=true#step:8:21)775[52](https://github.com/phylum-dev/cli/runs/6580368267?check_suite_focus=true#step:8:53)b.exe)

running 6 tests
test get_basic_status ... ignored
test get_job_status ... ignored
test get_job_status_non_existent_job ... ignored
test get_package_status ... ignored
test get_package_status_detailed ... ignored
test ping_system ... ignored
---TRIMMED---
```

This new `All features` step has almost the exact same output...just with the previously ignored tests getting run this time:

```sh
C:\Users\runneradmin\.cargo\bin\cargo.exe test --all-features
   Compiling phylum-cli v3.5.0 (D:\a\cli\cli\cli)
   Compiling xtask v0.1.0 (D:\a\cli\cli\xtask)
    Finished test [unoptimized + debuginfo] target(s) in 43.12s
     Running unittests src\lib.rs (target\debug\deps\phylum_cli-572328e7ef8dcd85.exe)

running 46 tests
test api::tests::create_client ... ok
test api::tests::get_job_status ... ok
---TRIMMED---
test types::tests::test_risk_level_ordering ... ok
test commands::parse::tests::it_can_identify_lock_file_types ... ok

test result: ok. 46 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.93s

     Running unittests src\bin\phylum.rs (target\debug\deps\phylum-debaa26529b4fdee.exe)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

     Running tests\cmd.rs (target\debug\deps\cmd-5824e46a56a0[11](https://github.com/phylum-dev/cli/runs/6580368267?check_suite_focus=true#step:9:12)de.exe)

running 6 tests
test get_job_status ... FAILED
test get_job_status_non_existent_job ... FAILED
test get_basic_status ... FAILED
test get_package_status ... FAILED
test get_package_status_detailed ... FAILED
test ping_system ... FAILED
---TRIMMED---
```

The overlap feels like waste...but maybe not...
Looking at the [`cargo test` documentation](https://doc.rust-lang.org/cargo/commands/cargo-test.html) did not yield any obvious command invocations to *only* run the normally ignored feature tests.

Reply:
Testing all features is not just about running tests that are only executed with certain features present, it's also about testing codepaths existing tests already covered that might behave differently with separate features present.

Consider a feature that changes API paths for example. We'd want to ensure that both old and new API paths work, but the tests that cover it would be the same.

Reply:
I'm slow sometimes...thanks for the longer explanation...I get it now!